lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -382,8 +383,19 @@ public static TopFieldCollector create(Sort sort, int numHits, int totalHitsThre
    * @return a {@link TopFieldCollector} instance which will sort the results by
    *         the sort criteria.
    */
+  public static TopFieldCollector create(Sort sort, int numHits, FieldDoc after, int totalHitsThreshold) {
+    if (totalHitsThreshold < 0) {
+      throw new IllegalArgumentException("totalHitsThreshold must be >= 0, got " + totalHitsThreshold);
+    }
+
+    return create(sort, numHits, after, HitsThresholdChecker.create(totalHitsThreshold));
+  }
+
+  /**
+   * Same as above with an additional parameter to allow passing in the threshold checker
+   */
   public static TopFieldCollector create(Sort sort, int numHits, FieldDoc after,
-      int totalHitsThreshold) {
+                                         HitsThresholdChecker hitsThresholdChecker) {
```

Comment:
I'd have a preference for not exposing `HitsThresholdChecker` in the user-facing API and instead providing users with a factory method for a collector manager that can be used with multiple threads, e.g. `public static CollectorManager<TopFieldCollector, TopDocs> createSharedManager(sort, numHits, after, totalHitsThreshold)`.